bridge: fix gitlab bridge when running multiple imports.

A-Hilaly opened the pull request:
fix an issue that was causing gitlab bridge to duplicate operations when running multiple imports

MichaelMure commented on bridge/gitlab/import.go:
```diff
@@ -141,23 +146,29 @@ func (gi *gitlabImporter) ensureNote(repo *cache.RepoCache, b *cache.BugCache, n
-				keyGitlabId: id,
-			},
-		)
+		if errResolve != nil {
```
Also, you can invert the logic of the if to keep the happy path on the left.

A-Hilaly replied:
Agreed that's much more better. I made the changes (inverted the logic). 
I still hesitate between `if errResolve == nil` and `if errResolve != cache.NoMatchingOpErr`. It's the only two cases we can have at this blocks

MichaelMure commented:
@cgivord have a look, this can help you if you run the importer more than once.

A-Hilaly commented:
I think i now where this is coming from. It will be probably fixed in #193 : https://github.com/MichaelMure/git-bug/pull/193/files#r312167031

Did you try to clean all local bugs and identities before trying to run the imports ? (`make clean-local-bugs` and `make clean-local-identities`)

A-Hilaly commented:
In our Gitlab test repository i run the import multiple times, here are my logs:

first import:
```
new identity: 478528d842a325a772c4bba4479a8a7f4484c7cd
new issue: 3ed2cd401ff4d8fc93e363134ef310e7e8be9774
new comment: 85b191b4ad4709be714381a27659f73687838504a38b78d9855b7ffa60c6ddd1
new comment: 5118bfbdd16864190bbe45159079140d07db59d9ce8e36baeee14bf844608fc5
new comment: 8bb5fd80bba7987b1367e0281337f6a9e91eb90cc4423d9abf37c18857af7d04
new comment: 41f9db0a682ebaf51b743eed84f043d9ec090f03f75c82bb692c6819ba9f83ad
new issue: 2f8536a98bd034bcf8954dbc557565c468147531
new comment: d641a1493022ab9023aa9ab8e32f15789089419cc28c5929f9f48f97001f23c1
new comment: f9f5d6fabcee47b6eb43fb3b5b11d61ebc82aa064781482e176bb04d2aab86aa
new identity: e74ca55146818c1750714941a7e48be0aa141e21
changed status: 05f94eb0641b45130798c767eb48788c05dadc7f5cab28c79863ca3bc7e692cd
changed status: 834ea7456e09fbddc5375bbf86f5bc47abb95bbdc57d54b2f94f0e6b3fbe8d3c
ignoring event: unsupported note type: 14
ignoring event: unsupported note type: 15
new issue: ffa63398c5f2253e4b80c27632e82e9d5a417fae
new comment: d7b2595d8779b640785c78e8897a4073892a1ce4008679ecf3839650f1a05fa7
new comment: cab4eb04ac81376b43498f6f3e5c2d73a7bee80154925ee96bf077c6f64ce1a1
new comment: 85362a4bd7affe157428372982a6a4d7feae83d621aa1c7e4c985085b349b60d
new comment: 7a8e1a25af9ae484b43916c3cbc3a0cc00f7746dbec11a169f0026f6cb353a54
changed title: b57147aa044befc155ee7be43a14928962688f3ad32d9b35ae1f2c4a876a3148
new issue: 4b185c3a9d6520c7783c136d57958b12e14f6c06
new comment: cfc5d52cf274c7884dc7a8342ef540c3c110a01c69e0646c210a0b6a19661c00
new comment: 6eeb8d2d24be165481cd12f005a0dce1f5cc7893fc7b9067d277153f6192b0d8
changed title: ffe3029a2c4ca539e7037286307c77b257024f97686e9f9de8859ed0d401cdf3
changed title: 9532bc8674d975d4ef200b273bf3fc0ac1aa100ab3d6c2bbecf0c019f0d09d27
new comment: 57cfdcb10a514997bb3b79f4cb67af2f8dbf51db5ace7380faabad4c9425bd4e
changed status: 97e95db1607850b30cac8838c15157286943c400b972fc5fa3db161b01bc097d
changed status: c2985ab892db064161d732a050571c217c777e3fd654cf5af44c836c455ecb9e
ignoring event: unsupported note type: 10
ignoring event: unsupported note type: 11
ignoring event: unsupported note type: 8
ignoring event: unsupported note type: 9
ignoring event: unsupported note type: 6
ignoring event: unsupported note type: 7
changed title: 21d5f5f68a2779ebec12183be364e1c4668f2c00a6c66a2efe687193716724d0
changed title: 496a6785f6228750c88e9c11e66f3067082612f6d0368882c4fe4cac7387a122
ignoring event: unsupported note type: 12
ignoring event: unsupported note type: 12
ignoring event: unsupported note type: 13
new comment: 7322b55e37b25d30ed5309375950839801d0ffe53763df167a87325afef5745a
new comment: 75efc8792902c27cabeb37a69a2aef1ea8d149d125ab718b909a0cf3f0f1a17f
new comment: 43c8e8743c59fca2316ef1538299301aec689a15d08cf21dfc08668bcb8c6289
new comment: 34b0c4bc235d9ecbe5aefa181cd5875827b19c75d09fd00a725ad49b32bd269a
new issue: 3815091e5d3270b49055b25ccc2f6dba67bdf6af
new comment: 3e3b18d5ad0a4354b27c619a3cdfc09e7be5cca8844e647e8cfc8cd8a16be249
changed title: d07a3b64cc45c837697ba7105b310f6bc724070fb0e30d27e7e454da40cb6132
Successfully imported 5 issues and 2 identities with test bridge
```

other imports:
```
ignoring event: bug already imported
ignoring event: bug already imported
ignoring event: unsupported note type: 14
ignoring event: unsupported note type: 15
ignoring event: bug already imported
ignoring event: bug already imported
ignoring event: unsupported note type: 10
ignoring event: unsupported note type: 11
ignoring event: unsupported note type: 8
ignoring event: unsupported note type: 9
ignoring event: unsupported note type: 6
ignoring event: unsupported note type: 7
ignoring event: unsupported note type: 12
ignoring event: unsupported note type: 12
ignoring event: unsupported note type: 13
ignoring event: bug already imported
Successfully imported 0 issues and 0 identities with test bridge
```

cgivord commented:
It seems to work after clearing git bug local caches !
Thanks !
